File: Intermediate-Algorithm-Scripting-Js-Py/1-Sum-All-Numbers-In-Range.js
// my solution
function sumAll(arr) {
    let low = Math.min(...arr)
    let high = Math.max(...arr)

    let total = 0
    for (let n = low; n <= high; n++) {
        total += n
    }
    return total
}

// using lodash
const range = require('lodash/range')

function sumAll(arr) {
    let low = Math.min(...arr)
    let high = Math.max(...arr)

    return range(low, high+1).reduce((a, b) => a + b)
}

File: Intermediate-Algorithm-Scripting-Js-Py/1-Sum-All-Numbers-In-Range.py
def sumAll(lst):
    low = min(lst)
    high = max(lst)

    return sum(range(low, high+1)) # high + 1 to make it inclusive

# === tests ===
print(sumAll([1, 4]))
print(sumAll([4, 1]))
print(sumAll([5, 10]))
print(sumAll([10, 5]))